text/2018-10-25-tikv-client-rust.md

```diff
@@ -0,0 +1,389 @@
+            &self,
+            ranges: Ranges,
+            each_limit: u32,
+        ) -> BatchScan<Self::AClient>
+        where
+            Ranges: AsRef<[Bounds]>,
+            Bounds: RangeBounds<Key>;
+
+        fn delete_range(&self, range: impl RangeBounds<Key>) -> DeleteRange<Self::AClient>;
+    }
+    ```
+
+#### Usage example of the Raw Key-Value API
+
+    ```rust
+    extern crate futures;
+    extern crate tikv_client;
+
+    use futures::future::Future;
+    use tikv_client::raw::Client;
+    use tikv_client::*;
+
+    fn main() {
+        let config = Config::new(vec!["127.0.0.1:3379"]);
+        let raw = raw::RawClient::new(&config)
+            .wait()
+            .expect("Could not connect to tikv");
+
+        let key: Key = b"Company".to_vec().into();
+        let value: Value = b"PingCAP".to_vec().into();
+
+        raw.put((Clone::clone(&key), Clone::clone(&value)))
+            .cf("test_cf")
+            .wait()
+            .expect("Could not put kv pair to tikv");
+        println!("Successfully put {:?}:{:?} to tikv", key, value);
+
+        let value = raw
+            .get(&key)
+            .cf("test_cf")
+            .wait()
+            .expect("Could not get value");
+        println!("Found val: {:?} for key: {:?}", value, key);
+
+        raw.delete(&key)
+            .cf("test_cf")
+            .wait()
+            .expect("Could not delete value");
+        println!("Key: {:?} deleted", key);
+
+        raw.get(&key)
+            .cf("test_cf")
+            .wait()
+            .expect_err("Get returned value for not existing key");
+
+        let keys = vec![b"k1".to_vec().into(), b"k2".to_vec().into()];
+
+        let _values = raw
+            .batch_get(&keys)
+            .cf("test_cf")
+            .wait()
+            .expect("Could not get values");
+
+        let start: Key = b"k1".to_vec().into();
+        let end: Key = b"k2".to_vec().into();
+        raw.scan(&start..&end, 10)
+            .cf("test_cf")
+            .key_only()
+            .wait()
+            .expect("Could not scan");
+
+        let ranges = [&start..&end, &start..&end];
+        raw.batch_scan(&ranges, 10)
+            .cf("test_cf")
+            .key_only()
+            .wait()
+            .expect("Could not batch scan");
+    }
+    ```
+
+The result is like:
+
+    ```bash
+    Successfully put Key([67, 111, 109, 112, 97, 110, 121]):Value([80, 105, 110, 103, 67, 65, 80]) to tikv
+    Found val: Value([80, 105, 110, 103, 67, 65, 80]) for key: Key([67, 111, 109, 112, 97, 110, 121])
+    Key: Key([67, 111, 109, 112, 97, 110, 121]) deleted
+    ```
+
+Raw Key-Value client is a client of the TiKV server and only supports the GET/BATCH_GET/PUT/BATCH_PUT/DELETE/BATCH_DELETE/SCAN/BATCH_SCAN/DELETE_RANGE commands. The Raw Key-Value client can be safely and concurrently accessed by multiple threads. Therefore, for one process, one client is enough generally.
+
+### Try the Transactional Key-Value API
```

Comment:
The previous section is called "Usage ... " and this section might follow the same style as well